src/pages/studyView/table/MutatedGenesTable.tsx

```diff
@@ -464,18 +392,20 @@ export class MutatedGenesTable extends React.Component<IMutatedGenesTablePros, {
     public render() {
         return (
             <div>
-                <MutatedGenesTableComponent
-                    width={this.props.width}
-                    height={this.props.height}
-                    data={this.props.promise.result || []}
-                    columns={this.tableColumns}
-                    showSelectSamples={true && this.preSelectedRows.length > 0}
-                    isSelectedRow={this.isSelectedRow}
-                    afterSelectingRows={this.afterSelectingRows}
-                    sortBy={this.sortBy}
-                    sortDirection={this.sortDirection}
-                    afterSorting={this.afterSorting}
-                />
+                {this.props.promise.isComplete && (
```

Comment:
maybe put the modal inside if statement too?

Reply:
We don't need to rerender the model when promise changes.